build

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env node
+
+if (!require('assert').strict) {
+	console.error("We require Node.js version 10 or later; you're using " + process.version);
+	process.exit(1);
+}
+
+var execSync = require('child_process').execSync;
+
+try {
+	require.resolve('sucrase');
+} catch (e) {
+	console.log('Installing dependencies...');
+	execSync('npm install', {stdio: 'inherit'});
+}
+
+execSync('npx sucrase ./sim -d ./sim --transforms typescript,imports');
```

Comment:
Is it possible to build to a different directory?

Reply:
Yes, of course. But on the bug you seemed to say you wanted it like this? What should it be instead?

Reply:
I kind of want to do `sim/` and `.sim-dist/` but imports would be kind of ugly.

Reply:
`.sim-dist/` is probably fine for now; it's not like we can't change it later. Although feel free to suggest alternatives if you don't like that scheme.

Reply:
The imports would look pretty fugly I think? What about my proposal for starting a new `ts` tree with `./src/` (the preferred pattern for typescript projects) putting `sim` under it (`./src/sim/`) and outputting to `./sim/`. As additional files get migrated over they would form a mirrored tree under `./src`, such that if the entire project were ever to be migrated it would have the same structure as today, just fully under `src` (at which point it would probably make sense to build to `./dist/`.

Reply:
I'm worried about users editing `./sim` directly, because they don't realize it's a built directory.

A lot of PS's tech decisions are based on the fact that a lot of people come in from basically zero programming background.

Reply:
Oh, I didn't notice `.` in `.sim-dist`. Good call. Will change.

Reply:
> Yes, of course. 

_Originally posted by @scheibo in https://github.com/Zarel/Pokemon-Showdown/pull/5210_

Yeah... turns out I'm wrong, this is the only setup which I can get to work. :(

Tried switching in 597f663 and changing the `require`s makes `tsc` fail the typecheck - it uses the `.ts` files for typechecking, so when the imports are changed to something else (be it `./src/sim/` or `.sim-dist/`) it no longer works.

Reply:
I'd suggest making it output `.d.ts` files, but I guess sucrase doesn't support that, either?